src/App.tsx

```diff
@@ -15,48 +16,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  Alphabet = 'Alphabet',
+  Length = 'Length',
+  Reverse = 'Reverse',
+  Default = '',
+}
+
+function getSortedPeople(
+  goods: string[],
+  sort: SortBy,
+  reverse: SortBy | boolean,
+): string[] {
+  let visibleGoods: string[] = [...goods];
+
+  switch (sort) {
+    case SortBy.Alphabet:
+      visibleGoods = visibleGoods.sort((
+        good1: string, good2: string,
+      ) => good1.localeCompare(good2));
+      break;
+    case SortBy.Length:
+      visibleGoods = visibleGoods.sort((
+        good1: string, good2: string,
+      ) => good1.length - good2.length);
+      break;
+    default:
+      visibleGoods = [...goodsFromServer];
+      break;
+  }
+
+  if (reverse) {
+    visibleGoods = visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortBy>(SortBy.Default);
+  const [reverseField, setReverseField] = useState<SortBy | boolean>(false);
```

Comment:
```suggestion
  const [isReversed, setIsReversed] = useState<boolean>(false);
```
Better make it a boolean as it can have only one of two value (reversed or not)